controllers/bucket_controller.go

```diff
@@ -107,6 +107,14 @@ func (r *BucketReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 
 	// Return early if the object is suspended.
 	if bucket.Spec.Suspend {
+		if r.MetricsRecorder != nil {
```

Comment:
Please see the correct way to implement this: https://github.com/fluxcd/kustomize-controller/pull/299